### PR TITLE
Add guides for upgrade from 1.4 and Helm

### PR DESCRIPTION
**Pull request by ostromart**
There have been questions in discuss about upgrading from Helm and/or 1.4. This guide provides instruction for those cases.

### CODE REVIEW

File: content/en/docs/setup/upgrade/index.md [hunk 1]
```diff
@@ -190,3 +196,58 @@ the process will restore Istio back to the Istio version that was installed befo
...
+      containers:
+      - command:
+        ...
+        - --log_output_level=default:info
+        - --enable-validation=false
+{{< /text >}}
+
+Save and quit the editor to update the deployment configuration in the cluster.
+
+1. Remove the ValidatingWebhookConfiguration Custom Resource (CR) with the following command:
```
Comment (frankbu) [inline content/en/docs/setup/upgrade/index.md:249]:
```suggestion
1. Remove the `ValidatingWebhookConfiguration` Custom Resource (CR) with the following command:
```

File: content/en/docs/setup/upgrade/index.md [hunk 2]
```diff
@@ -190,3 +196,58 @@ the process will restore Istio back to the Istio version that was installed befo
 `istioctl install` also installs the same Istio control plane, but does not
 perform any checks. For example, default values applied to the cluster for a configuration
 profile may change without warning.
+
+## Upgrading from Helm installations
+
+For Istio installations done with Helm, the canary upgrade process must be used.
+The canary control plane must be installed with an `IstioOperator` CR equivalent
+to the Helm `values.yaml`used to install Istio. The `istioctl manifest migrate` command (using `istioctl` v1.6)
```
Comment (frankbu) [inline content/en/docs/setup/upgrade/index.md:204]:
```suggestion
to the Helm `values.yaml` used to install Istio. The `istioctl manifest migrate` command (using `istioctl` v1.6)
```

File: content/en/docs/setup/upgrade/index.md [hunk 3]
```diff
@@ -12,10 +12,16 @@ Upgrading Istio can be done by first running a canary deployment of the new cont
...
+at the same time. A canary version of an upgrade can be started by installing the new Istio version's control plane
+next to the old one, using a different `revision` setting. Each revision is a full Istio control plane implementation
+with its own `Deployment`, `Service`, etc.
 
-### Control plane
+See additional notes for [upgrading from Helm installations](#upgrading-from-helm-installations)
+and [upgrading from 1.4.x](#upgrading-from-1.4).
 
+### Control plane
+
```
Comment (frankbu) [inline content/en/docs/setup/upgrade/index.md:24]:
```suggestion
### Control plane

```

File: content/en/docs/setup/upgrade/index.md [hunk 4]
```diff
@@ -190,3 +196,58 @@ the process will restore Istio back to the Istio version that was installed befo
 `istioctl install` also installs the same Istio control plane, but does not
 perform any checks. For example, default values applied to the cluster for a configuration
 profile may change without warning.
+
+## Upgrading from Helm installations
+
+For Istio installations done with Helm, the canary upgrade process must be used.
+The canary control plane must be installed with an `IstioOperator` CR equivalent
+to the Helm `values.yaml`used to install Istio. The `istioctl manifest migrate` command (using `istioctl` v1.6)
+simplifies the migration by automatically translating `values.yaml` to IstioOperator CR format.
```
Comment (frankbu) [inline content/en/docs/setup/upgrade/index.md:205]:
```suggestion
simplifies the migration by automatically translating `values.yaml` to IstioOperator CR format.
```

File: content/en/docs/setup/upgrade/index.md [hunk 5]
```diff
@@ -190,3 +196,58 @@ the process will restore Istio back to the Istio version that was installed befo
...
+{{< text bash >}}
+$ istioctl manifest migrate <path-to-values.yaml> > iop.yaml
+{{< /text >}}
+
+Inspect the generated `iop.yaml` file to ensure it's correct. You can use this CR to install a 1.6 Istio canary
+control plane with the same settings as the Helm installed control plane.
+
+## Upgrading from 1.4
+
+Migrating from 1.4 Istio (installed with `istioctl` or Helm) is similar to the process for 1.5 using canary, with one
```
Comment (frankbu) [inline content/en/docs/setup/upgrade/index.md:217]:
```suggestion
Migrating from 1.4 Istio (installed with `istioctl` or Helm) is similar to the process for 1.5 using canary, with one
```

File: content/en/docs/setup/upgrade/index.md [hunk 6]
```diff
@@ -190,3 +196,58 @@ the process will restore Istio back to the Istio version that was installed befo
...
+{{< /text >}}
+
+Inspect the generated `iop.yaml` file to ensure it's correct. You can use this CR to install a 1.6 Istio canary
+control plane with the same settings as the Helm installed control plane.
+
+## Upgrading from 1.4
+
+Migrating from 1.4 Istio (installed with `istioctl` or Helm) is similar to the process for 1.5 using canary, with one 
+additional step. Istio 1.4 validation does not recognize some 1.6 resources, and the 1.4 validation webhook prevents
+Istio 1.6 from functioning correctly.
```
Comment (frankbu) [inline content/en/docs/setup/upgrade/index.md:219]:
```suggestion
Istio 1.6 from functioning correctly.
```

File: content/en/docs/setup/upgrade/index.md [hunk 7]
```diff
@@ -190,3 +196,58 @@ the process will restore Istio back to the Istio version that was installed befo
 `istioctl install` also installs the same Istio control plane, but does not
 perform any checks. For example, default values applied to the cluster for a configuration
 profile may change without warning.
+
+## Upgrading from Helm installations
+
+For Istio installations done with Helm, the canary upgrade process must be used.
+The canary control plane must be installed with an `IstioOperator` CR equivalent
+to the Helm `values.yaml` used to install Istio. The `istioctl manifest migrate` command (using `istioctl` v1.6)
```
Comment (frankbu) [inline content/en/docs/setup/upgrade/index.md:204]: One more lint error.
```suggestion
to the Helm `values.yaml` used to install Istio. The `istioctl manifest migrate` command (using `istioctl` 1.6)
```

**Comment by ostromart** [comment]
Thanks Frank and sorry for the linter problems. I was going to run through it after any high level comments but you beat me to it.

**Pull request merged**